### PR TITLE
Expose gop

### PR DESCRIPTION
**Pull request by krystalmejia24**

### CODE REVIEW

File: provider/bitmovin/internal/configuration/codec/h264.go [hunk 1]
```diff
@@ -76,12 +76,25 @@ func h264ConfigFrom(preset db.Preset) (model.H264VideoConfiguration, error) {
 
 	presetGOPSize := preset.Video.GopSize
 	if presetGOPSize != "" {
-		gopSize, err := gopSizeFrom(presetGOPSize)
-		if err != nil {
-			return model.H264VideoConfiguration{}, err
+		switch strings.ToLower(preset.Video.GopUnit) {
+		case "frames", "":
```
[the remark this answers is not in the document]
Comment (krystalmejia24) [reply provider/bitmovin/internal/configuration/codec/h264.go]: @zsiec what would be the best way to do that? add those string values to our model?

File: provider/bitmovin/internal/configuration/codec/h264.go [hunk 2]
```diff
@@ -76,12 +76,25 @@ func h264ConfigFrom(preset db.Preset) (model.H264VideoConfiguration, error) {
...
+			}
+			cfg.MinGop = gopSize
+			cfg.MaxGop = gopSize
+		case "seconds":
+			gopSize, err := keyIntervalFrom(presetGOPSize)
+			if err != nil {
+				return model.H264VideoConfiguration{}, err
+			}
+			cfg.MinKeyframeInterval = gopSize
+			cfg.MaxKeyframeInterval = gopSize
```
[the remark this answers is not in the document]
Comment (krystalmejia24) [reply provider/bitmovin/internal/configuration/codec/h264.go]: @zsiec if you think its useful. we do have validator on mediahub side throwing those errors if the `gopUnit` is invalid.

File: provider/hybrik/hybrik_test.go [hunk 3]
```diff
@@ -702,8 +698,6 @@ func TestHybrikProvider_presetsToTranscodeJob(t *testing.T) {
 															Codec:          "h265",
 															Profile:        "main10",
 															Tune:           "grain",
-															MinGOPFrames:   120,
-															MaxGOPFrames:   120,
 															ExactGOPFrames: 120,
 															InterlaceMode:  "progressive",
 															ChromaFormat:   "yuv420p10le",
```
[the remark this answers is not in the document]
Comment (krystalmejia24) [reply provider/hybrik/hybrik_test.go]: i agree. will do that and resolve this when i get that done

File: provider/bitmovin/internal/configuration/codec/h264.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (zsiec) [reply provider/bitmovin/internal/configuration/codec/h264.go]: We could have them live close to the definition of the structures that carry them. We could add here: https://github.com/cbsinteractive/video-transcoding-api/blob/master/db/types.go

something like:

```go
type GOPUnit = string

const (
    GOPUnitSeconds GOPUnit = "seconds"
    GOPUnitFrames GOPUnit = "frames"
)
```

(P.S. in general, I'm not thrilled that we're using the models defined in the `db` package everywhere directly. It couples things a bit too much, I feel this is something we'll want to address in a coming refactor)

File: provider/bitmovin/internal/configuration/codec/h264.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (zsiec) [reply provider/bitmovin/internal/configuration/codec/h264.go]: That's true although I feel like we should force ourselves to think about these services outside the context of how we're using them to ensure they maintain their usability in isolation. (Tomorrow we might have some other service talking to this API that doesn't have all the validation logic we have in mediahub today)

File: provider/bitmovin/internal/configuration/codec/h264.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (krystalmejia24) [reply provider/bitmovin/internal/configuration/codec/h264.go]: perfect. i named it the exact same thing! will push the update

**Pull request merged**